scripts/src/checkprcontent/checkpr.py

```diff
@@ -10,10 +10,73 @@
 except ImportError:
     from yaml import Loader, Dumper
 
+sys.path.append('../')
+from owners import owners_file
+from report import verifier_report
 
 ALLOW_CI_CHANGES = "allow/ci-changes"
 TYPE_MATCH_EXPRESSION = "(partners|redhat|community)"
 
+def check_provider_delivery(report_in_pr,num_files_in_pr,report_file_match):
+
+    print(f"[INFO] report in PR {report_in_pr}")
+    print(f"[INFO] num files in PR {num_files_in_pr}")
+    
+    category, organization, chart, version = report_file_match.groups()
+
+    print(f"read owners file : {category}/{organization}/{chart}" )
+    found_owners,owner_data = owners_file.get_owner_data(category, organization, chart)
+
+    if found_owners:
+        owner_provider_delivery = owners_file.get_provider_delivery(owner_data)
+        print(f"[INFO] providerDelivery from OWNERS : {owner_provider_delivery}")
+    else:
+        msg = "[ERROR] OWNERS file was not found."
+        print(msg)
+        print(f"::set-output name=owners-error-message::{msg}")
+        sys.exit(1)
+
+    if report_in_pr:
+        report_file_path = os.path.join("pr-branch","charts", category, organization, chart, version, "report.yaml")
+        print(f"read report file : {report_file_path}" )
+        found_report,report_data = verifier_report.get_report_data(report_file_path)
+
+        if found_report:
+            report_provider_delivery = verifier_report.get_provider_delivery(report_data)
+            print(f"[INFO] providerDelivery from report : {report_provider_delivery}")
+        else:
+            msg = f"[ERROR] Failed tp open report: {report_file_path}."
+            print(msg)
+            print(f"::set-output name=pr-content-error-message::{msg}")
+            sys.exit(1)
+
+    if report_in_pr and num_files_in_pr > 1:
+        if report_provider_delivery or owner_provider_delivery:
+            msg = f"[ERROR] OWNERS file and/or report indicate provider controlled delivery but pull request is not report only."
+            print(msg)
+            print(f"::set-output name=pr-content-error-message::{msg}")
+            sys.exit(1)
+    elif report_in_pr:
+        if report_provider_delivery and owner_provider_delivery:
+            print(f"[INFO] providerDelivery is a go")
+            print(f"::set-output name=providerDelivery::True")
+        elif report_provider_delivery:
+            msg = f"[ERROR] Report indicates provider controlled delivery but OWNERS file does not."
```

Comment:
I would only use the f-strings when I'm parametrizing the string.

Reply:
will update later.